### DISCUSSION TITLE
Upgrading from 3.4.6 to 4.1.3 - any middle jump?

### DISCUSSION THREAD

**Discussion by gblnunes** [netbox-community/netbox#17677] [Q&A] [answered]
Hello Community!

This is the first time I have had to update Netbox and I going from 3.4.6 to 4.1.3. Is there any attention I should have? Is there any middle version I should go for before going to 4.1.3? We use an external instance of PostgreSQL, any specific procedure to execute?

Thanks for the help!

**Comment by candlerb** [comment, marked as answer]
Yes, you must go via 3.7.x (3.7.8 preferred)

https://github.com/netbox-community/netbox/discussions/17432#discussioncomment-10606668

Upgrade instructions [here](https://netboxlabs.com/docs/netbox/en/stable/installation/upgrading/). Depending on the versions of OS, python and postgres you have in your current system, it may be better to create a fresh VM (and this is certainly the safest approach). That is:

* build a new v3.7.8 instance
* [import](https://netboxlabs.com/docs/netbox/en/stable/administration/replicating-netbox/) your database into it and then run `./upgrade.sh`
* upgrade to v4.1.3

**Reply by gblnunes** [reply]
Thanks for the explanation! I will follow this approach!

**Reply by candlerb** [reply]
> We use an external instance of PostgreSQL

I forgot to add: you'll need to make sure that is running PostgreSQL v12 or later.

https://netboxlabs.com/docs/netbox/en/stable/installation/#requirements
